shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -107,18 +106,50 @@
         @Override
         public void createAndroidViewForPlatformView(
             @NonNull PlatformViewsChannel.PlatformViewCreationRequest request) {
-          // API level 19 is required for android.graphics.ImageReader.
+          // API level 19 is required for `android.graphics.ImageReader`.
           ensureValidAndroidVersion(Build.VERSION_CODES.KITKAT);
-          platformViewRequests.put(request.viewId, request);
+
+          if (request == null) {
+            throw new IllegalStateException(
+                "Platform view hasn't been initialized from the platform view channel.");
+          }
+
+          if (!validateDirection(request.direction)) {
+            throw new IllegalStateException(
+                "Trying to create a view with unknown direction value: "
+                    + request.direction
+                    + "(view id: "
+                    + request.viewId
+                    + ")");
+          }
+
+          final PlatformViewFactory factory = registry.getFactory(request.viewType);
+          if (factory == null) {
+            throw new IllegalStateException(
+                "Trying to create a platform view of unregistered type: " + request.viewType);
+          }
+
+          Object createParams = null;
+          if (request.params != null) {
+            createParams = factory.getCreateArgsCodec().decodeMessage(request.params);
+          }
+
+          final PlatformView platformView = factory.create(context, request.viewId, createParams);
+          final View view = platformView.getView();
+          if (view == null) {
+            throw new IllegalStateException(
+                "PlatformView#getView() returned null, but an Android view reference was expected.");
+          }
+          if (view.getParent() != null) {
+            throw new IllegalStateException(
```

Comment:
Who's responsible for handling all these exceptions?  Will they propagate all the way up and crash the app?  If so, is that the desired behavior, or should they translate to Dart "soft" exceptions?  Can they ever happen in the normal operation of an app, or do they signal developer error?

Reply:
It will be a `Future` error in https://github.com/flutter/flutter/blob/f512e86c2a210d8402e43a3352c1b2fcf4308a43/packages/flutter/lib/src/services/platform_views.dart#L749, which will make the app crash with this error message.

It's something that should be addressed in the platform code by the developer before the API can be used in Dart.